npm_and_yarn/spec/dependabot/npm_and_yarn/file_parser/lockfile_parser_spec.rb

```diff
@@ -70,6 +70,68 @@
       end
     end
 
+    context "for pnpm lockfiles" do
+      let(:dependency_files) { project_dependency_files("pnpm/only_dev_dependencies") }
+
+      it "parses the dependencies" do
+        expect(dependencies.map(&:name)).to contain_exactly("etag")
+      end
+
+      context "that contains an empty version string" do
+        let(:dependency_files) { project_dependency_files("pnpm/empty_version") }
+
+        it "raises a DependencyFileNotParseable error" do
+          expect { dependencies }.
+            to raise_error(Dependabot::DependencyFileNotParseable) do |error|
+              expect(error.file_name).to eq("pnpm-lock.yaml")
+            end
+        end
+      end
+
+      context "that contains an aliased dependency" do
+        let(:dependency_files) { project_dependency_files("pnpm/aliased_dependency") }
+
+        it "excludes the dependency" do
+          # Lockfile contains 11 dependencies but one is an alias
+          expect(dependencies.count).to eq(10)
+          expect(dependencies.map(&:name)).to_not include("my-fetch-factory")
+        end
+      end
+
+      context "that contain multiple dependencies" do
+        let(:dependency_files) { project_dependency_files("pnpm/no_lockfile_change") }
+
+        its(:length) { is_expected.to eq(370) }
+
+        describe "a repeated dependency" do
+          subject { dependencies.find { |d| d.name == "async" } }
+
+          its(:version) { is_expected.to eq("1.5.2") }
+        end
+      end
+
+      context "that contains specified locked to versions with peer disambiguation suffix" do
```

Comment:
typo? reads like there's a missing word, but I'm actually not sure what it might be...

Reply:
Yes, I think I meant

```suggestion
      context "locked to versions with peer disambiguation suffix" do
```

Will update!